src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/nd4jutils/Nd4jIOUtils.java

```diff
@@ -0,0 +1,161 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.nd4jutils;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+/**
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class Nd4jIOUtils {
+
+    /**
+     * Write NDArray to a binary dump
+     *
+     * @param arr an instance of {@link INDArray}
+     * @param outputFile the output path
+     */
+    public static void writeNDArrayToBinaryDumpFile(@Nonnull final INDArray arr,
```

Comment:
They are not used now (I used them before though). We may have a use case for them in the future since dumping NDArrays to binary is _much_ master than creating a human-readable tsv. I'd rather keep them in this class for possible future use especially as they have unit tests.